mgradm/cmd/backup/backup.go

```diff
@@ -0,0 +1,136 @@
+// SPDX-FileCopyrightText: 2025 SUSE LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package backup
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+
+	"github.com/uyuni-project/uyuni-tools/mgradm/cmd/backup/create"
+	"github.com/uyuni-project/uyuni-tools/mgradm/cmd/backup/restore"
+	"github.com/uyuni-project/uyuni-tools/mgradm/cmd/backup/shared"
+	. "github.com/uyuni-project/uyuni-tools/shared/l10n"
+	"github.com/uyuni-project/uyuni-tools/shared/types"
+	"github.com/uyuni-project/uyuni-tools/shared/utils"
+)
+
+func newCreateCmd(globalFlags *types.GlobalFlags, run utils.CommandFunc[shared.Flagpole]) *cobra.Command {
+	var flags shared.Flagpole
+
+	createCmd := &cobra.Command{
+		Use:   "create output-directory",
+		Args:  cobra.ExactArgs(1),
+		Short: L("Create backup"),
+		Long:  L("Create backup of the already configured Uyuni system"),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return utils.CommandHelper(globalFlags, cmd, args, &flags, nil, run)
+		},
+	}
+
+	createCmd.Flags().StringSlice("skipvolumes", []string{}, L("Skip backup of selected volumes"))
+	createCmd.Flags().StringSlice("extravolumes", []string{}, L("Backup additional volumes to the build-in ones"))
+	createCmd.Flags().Bool("skipdatabase", false, L("Do not backup database volume, allow online backup."))
+	createCmd.Flags().Bool("skipimages", false, L("Do not backup container images"))
+	createCmd.Flags().Bool("skipconfig", false, L("Do not backup podman configuration. On restore defaults will be used"))
+	createCmd.Flags().Bool("norestart", false, L("Do not restart services after backup is done"))
+	createCmd.Flags().Bool("dryrun", false, L("Print expected actions, but no action is done"))
+
+	return createCmd
+}
+
+func newRestoreCmd(globalFlags *types.GlobalFlags, run utils.CommandFunc[shared.Flagpole]) *cobra.Command {
+	var flags shared.Flagpole
+
+	restoreCmd := &cobra.Command{
+		Use:   "restore directory",
+		Args:  cobra.ExactArgs(1),
+		Short: L("Restore backup from the directory"),
+		Long:  L("Restore backup of the previously configured Uyuni system from a specified directory"),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return utils.CommandHelper(globalFlags, cmd, args, &flags, nil, run)
+		},
+	}
+
+	restoreCmd.Flags().StringSlice("skipvolumes", []string{}, L("Skip restore of selected volumes"))
+	restoreCmd.Flags().Bool("skipdatabase", false, L("Do not restore database volume"))
```

Comment:
just a nitpick, maybe it'd be nice to have flags like `skip-database`. I'm thinking at the scenario where a configuration yaml file is used, in this way we can have a single `skip:` section